src/ORM/EagerLoader.php

```diff
@@ -126,14 +126,23 @@ class EagerLoader
      * @param array|string $associations list of table aliases to be queried.
      * When this method is called multiple times it will merge previous list with
      * the new one.
+     * @param callable|null $queryBuilder The query builder callable
      * @return array Containments.
      */
-    public function contain($associations = [])
+    public function contain($associations = [], callable $queryBuilder = null)
     {
         if (empty($associations)) {
             return $this->_containments;
         }
 
+        if ($queryBuilder) {
+            $associations = [
+                $associations => [
```

Comment:
What happens when the parameter is an array? Someone will goof that up. I think you want line 138 to be `if ($queryBuilder && is_string($associations)) {`